src/admin-views/category-colors/add-category-fields.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Template for adding category colors in the "Add New Category" form.
+ *
+ * @var array  $category_colors An associative array of meta keys and their values.
+ * @var object $taxonomy        The taxonomy object.
+ */
+
+?>
+
+<div class="tec_category_colors__wrap">
+	<h2><?php esc_html_e( 'Category Colors', 'the-events-calendar' ); ?></h2>
+	<?php wp_nonce_field( 'save_category_colors', 'tec_category_colors_nonce' ); ?>
+	<div class="tec-events-category-colors__container">
+		<div class="tec-events-category-colors__grid">
+			<div class="form-field">
+				<label for="tec-events-category-colors__primary"><?php esc_html_e( 'Primary Color', 'the-events-calendar' ); ?></label>
+				<input
+					type="text"
+					id="tec-events-category-colors__primary"
+					name="tec_events_category-color[primary]"
```

Comment:
Use either all dash or all underscore. I prefer dash but 🤷🏼 
BEM is only applied to classes. 
(not to IDs either, I see you up there "tec-events-category-colors__primary")